Add resin JMX metrics integration (#512)
DataDog/integrations-extras: resin/README.md

# Agent Check: Resin

## Overview

This check monitors [Resin][1] through the Datadog Agent.

## Setup

### Installation

The Resin check is not included in the [Datadog Agent][2] package, so you will
need to install it yourself.

### Configuration

1. Edit the `resin.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your resin performance data. See the [sample resin.d/conf.yaml][2] for all available configuration options.

2. [Restart the Agent][3].

### Validation

[Run the Agent's status subcommand][4] and look for `resin` under the Checks section.

## Data Collected

### Metrics

> Resin does not include any metrics.

Reviewer: ```suggestion
See [metadata.csv][6] for a list of metrics provided by this integration.
```

### Service Checks

> Resin does not include any service checks.

Reviewer: Please describe the service checks here

### Events

Resin does not include any events.

## Troubleshooting

Need help? Contact [Datadog support][5].

> [1]: **LINK_TO_INTEGERATION_SITE**

Reviewer: ```suggestion
[1]: https://caucho.com/
```

[2]: https://github.com/DataDog/integrations-core/blob/master/resin/datadog_checks/resin/data/conf.yaml.example
[3]: https://docs.datadoghq.com/agent/guide/agent-commands/?tab=agentv6#start-stop-and-restart-the-agent
[4]: https://docs.datadoghq.com/agent/guide/agent-commands/?tab=agentv6#agent-status-and-information

> [5]: https://docs.datadoghq.com/help

Reviewer: ```suggestion
[5]: https://docs.datadoghq.com/help
[6]: https://github.com/DataDog/integrations-extras/blob/master/resin/metadata.csv
```

# Agent Check: Resin

## Overview

This check monitors [Resin][1] through the Datadog Agent.

## Setup

### Installation

The Resin check is not included in the [Datadog Agent][2] package, so you will
need to install it yourself.

> ### Configuration

Reviewer: I'm reading that you need to add `<server-default><jvm-arg>-Dcom.sun.management.jmxremote</jvm-arg></server-default>` in `resin.xml` to the clusters you want to monitor. Can you add a link to the proper documentation here?

Author: That might be true.  We currently use this with the APM agent which has the embedded JMXfetch that connects locally.

Reviewer: Yes, that's what it says you need to connect locally. To connect remotely it needs a different option.

Author: I can confirm that you do not need that option if you are connecting via APM.  These were the options that we currently use without any jmx options and we still receive metrics. 

```
<jvm-arg>-Ddd.jmxfetch.enabled=true</jvm-arg>
<jvm-arg>-Ddd.jmxfetch.metrics-configs=datadog/jmx.resin.yaml</jvm-arg>
<jvm-arg>-Ddd.jmxfetch.refresh-beans-period=120</jvm-arg>
```

1. Edit the `resin.d/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your resin performance data. See the [sample resin.d/conf.yaml][2] for all available configuration options.

2. [Restart the Agent][3].

### Validation

[Run the Agent's status subcommand][4] and look for `resin` under the Checks section.

## Data Collected

### Metrics

See [metadata.csv][6] for a list of metrics provided by this integration.

### Service Checks

> Resin instance is running and reporting JVM metrics

Reviewer: ```suggestion
**resin.can_connect**:

Returns `CRITICAL` if the Agent is unable to connect to and collect metrics from the monitored Resin instance. Returns `OK` otherwise.
```
